### PR TITLE
Migrate from UDS to DRO

### PR DESCRIPTION
**Pull request by yuvraj-vansure**
This epic is basically a requirement to continue MAS's use of metrics and metering. UDS is getting sunset in approximately 2Q 2024, and we need a replacement technology. IBM Software Central has stepped up to the plate and will be extending the scope of the RHM operator to do this. This will be released as the "Data Reporter Operator", or DRO.

https://jsw.ibm.com/browse/MASCORE-16

### CODE REVIEW

File: ibm/mas_devops/roles/dro/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+dro_action: "{{ lookup('env', 'DRO_ACTION') | default('install', true) }}"
+dro_version: "{{ lookup('env', 'DRO_VERSION') | default('', true) }}"
+
+# DRO in-cluster installation
+# -----------------------------------------------------------------------------
+dro_storage_class: "{{ lookup('env', 'DRO_STORAGE_CLASS') }}"
```
Comment (Ashish-Mahawal1) [inline ibm/mas_devops/roles/dro/defaults/main.yml:7]: This var is missing in readme.md

File: ibm/mas_devops/roles/dro/tasks/uninstall/main.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+# Delete secrets
+- name: "uninstall : Delete the ibm-data-reporter-operator-api SA token secret"
+  kubernetes.core.k8s:
+    state: absent
+    api_version: v1
+    kind: Secret
+    namespace: redhat-marketplace
+    name: ibm-data-reporter-operator-api-token
+
+- name: "uninstall : Delete the imarketplace secret"
```
Comment (Ashish-Mahawal1) [inline ibm/mas_devops/roles/dro/tasks/uninstall/main.yml:122]: some typo `imarketplace`

File: ibm/mas_devops/roles/dro/tasks/install/main.yml [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+# -----------------------------------------------------------------------------
+- name: Check if operator group is present in redhat-marketplace already
+  kubernetes.core.k8s_info:
+    namespace: redhat-marketplace
+    kind: OperatorGroup
+  register: og_info
+
+- name: "Create OperatorGroup in redhat-marketplace namespace if not exist"
+  kubernetes.core.k8s:
+    definition: "{{ lookup('template', 'templates/operatorgroup.yml.j2') }}"
```
Comment (Ashish-Mahawal1) [inline ibm/mas_devops/roles/dro/tasks/install/main.yml:40]: i think a `when` condition is required to skip this step if operator group is present

File: ibm/mas_devops/roles/dro/tasks/install/main.yml [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+# Create Marketplace Pull Secret
+- name: Get marketplace secret
```
Comment (Ashish-Mahawal1) [inline ibm/mas_devops/roles/dro/tasks/install/main.yml:43]: This task is to check whether the secret is already created or not .If yes then we are not using `rm_sec` info.
I think we can use this info in task `Create Secret redhat-marketplace-pull-secret` and in `check pull secret` where we don't fail if secret has been already created .

File: ibm/mas_devops/roles/dro/tasks/install/main.yml (hunk 3 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/dro/tasks/install/main.yml:37]: operatorgroup.yml.j2 have the step to check if operator group is present.

File: ibm/mas_devops/roles/dro/tasks/install/main.yml (hunk 4 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/dro/tasks/install/main.yml:40]: Updated conditions of mentioned tasks to use `rm_sec` check

File: ibm/mas_devops/roles/dro/tasks/uninstall/main.yml (hunk 2 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/dro/tasks/uninstall/main.yml]: fixed the typo

**Comment by lokesh-sreedhara** [comment]
image

can u add dro role into oneclick_core.yml ? would be good have both in a if loop controlled by UDS_ACTION or DRO_ACTION variables

**Comment by HariPalleti** [comment]
@yuvraj-vansure Don't it needs a new environment variable like `BAS_INSTALL` for the customer to select which one to install `DRO` or `UDS`

**Comment by yuvraj-vansure** [comment]
closing this as created new PR #1049

**Pull request closed without merging**